components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImplCopy.java

```diff
@@ -0,0 +1,362 @@
+                return response;
+            } catch (UMAClientException e) {
+                handleErrorResponse(e, false);
+                log.error("Invalid request.Request with valid resource Id to update the resource. ", e);
+            } catch (UMAServiceException e) {
+                handleErrorResponse(e, true);
+            }
+            return response;
+        } else {
+            response = Response.status(Response.Status.UNAUTHORIZED).build();
+        }
+        return response;
+    }
+
+    /**
+     * Update a resource
+     *
+     * @param updatedResource details of the resource to be updated
+     * @param resourceId      ID of the resource to be updated
+     * @return Response with the updated resource
+     */
+    @Override
+    public Response updateResource(String resourceId, ResourceDetailsDTO updatedResource, MessageContext context) {
+
+        Response response = null;
+        if (!isValidateToken(context)) {
+            try {
+                if (isResourceIdValid(resourceId)) {
+
+                    try {
+                        Resource resourceRegistration = ResourceUtils.getResourceService().updateResource(resourceId,
+                                ResourceUtils.
+                                        getResource(updatedResource));
+                        resourceRegistration.setResourceId(resourceId);
+                        UpdateResourceDTO updateResourceDTO = ResourceUtils.updateResponse(resourceRegistration);
+                        return Response.status(Response.Status.CREATED).entity(updateResourceDTO).build();
+                    } catch (UMAServiceException e) {
+                        handleErrorResponse(e, true);
+                        log.error("Invalid request.Request with valid resource Id to update the resource. ", e);
+
+                    } catch (Throwable throwable) {
+                        handleErrorResponse((UMAException) throwable, true);
+                        log.error("Internal server error occurred. ", throwable);
+                    }
+                } else {
+                    throw new UMAClientException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID);
+                }
+            } catch (UMAClientException e) {
+                handleErrorResponse(e, false);
+                log.error("Client error when retrieving resource from the resource server.", e);
+            }
+            return response;
+        } else {
+            response = Response.status(Response.Status.UNAUTHORIZED).build();
+        }
+        return response;
+    }
+
+    /**
+     * Delete the resource for the given resourceId
+     *
+     * @param resourceId resourceId of the resource which need to get deleted
+     * @return Response with the status of resource deletion
+     */
+    @Override
+    public Response deleteResource(String resourceId, MessageContext context) {
+
+        Response response = null;
+        if (!isValidateToken(context)) {
```

Comment:
This logic seems not meaningful. Why are we going in to this block if the token is not valid ? Or else change the name of the method to validateToken and throw an exception if validation fails